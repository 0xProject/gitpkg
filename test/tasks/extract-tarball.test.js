import path from 'path';
import del from 'del';
import folderHash from 'folder-hash';
import extractTarball from '../../src/tasks/Publish/extract-tarball';

const extractPath = path.resolve(__dirname, '../fixtures/simple-package');

describe('while using extractTarball()', () => {
  beforeAll(() => del(extractPath));

  it('should extract tarball successfully', async () => {
    const tarballPath = path.resolve(__dirname, '../fixtures/simple-package-1.0.0.tgz');

    await extractTarball(tarballPath, extractPath);

    const hash = await folderHash.hashElement(extractPath);
    expect(hash).toMatchSnapshot();
  });

  it.skip('should throw error with invalid tarball path', async () => {
    try {
      const tarballPath = path.resolve(__dirname, '../fixtures/doesnt-exist-1.0.0.tgz');
      await extractTarball(tarballPath, extractPath);
    } catch (err) {
      expect(() => {
        throw err;
      }).toThrowErrorMatchingSnapshot();
    }
  });
});
