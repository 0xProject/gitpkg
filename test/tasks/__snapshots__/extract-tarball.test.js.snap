// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while using extractTarball() should extract tarball successfully 1`] = `
HashedFolder {
  "children": Array [
    HashedFolder {
      "children": Array [
        HashedFolder {
          "children": Array [
            HashedFile {
              "hash": "8vwKhXYbbUHm5soiQgmXvw/kC+k=",
              "name": "index.js",
            },
          ],
          "hash": "rJQ8De30VGH1jlaMI7WDzlgrXIs=",
          "name": "lib",
        },
        HashedFile {
          "hash": "gon524qOo+f85zDzzz7PdjzJbDI=",
          "name": "package.json",
        },
      ],
      "hash": "J6KNlLO0sZjT3nb8KisV5AXNzMs=",
      "name": "package",
    },
  ],
  "hash": "4jqRdapZPHxsyk6zsSTxlHXWOII=",
  "name": "simple-package",
}
`;

exports[`while using extractTarball() should throw error with invalid tarball path 1`] = `"ENOENT: no such file or directory, open '/Users/ramasilveyra/dev/personal/gitpkg/test/fixtures/doesnt-exist-1.0.0.tgz'"`;
